exec/java-exec/src/test/java/org/apache/drill/exec/rpc/user/security/TestUserBitSSL.java

```diff
@@ -0,0 +1,338 @@
+    final Properties connectionProps = new Properties();
+    connectionProps.setProperty(DrillProperties.ENABLE_TLS, "true");
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PATH, tsPath);
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PASSWORD, "bad_password");
+    connectionProps.setProperty(DrillProperties.DISABLE_HOST_VERIFICATION, "true");
+    boolean failureCaught = false;
+    try {
+      updateClient(connectionProps);
+    } catch (Exception e) {
+      failureCaught = true;
+    }
+    assertEquals(failureCaught, true);
+  }
+
+  @Test
+  public void testSSLConnectionFailEmptyTrustStore() throws Exception {
+    final Properties connectionProps = new Properties();
+    connectionProps.setProperty(DrillProperties.ENABLE_TLS, "true");
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PATH, emptyTSPath);
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PASSWORD, "drill123");
+    connectionProps.setProperty(DrillProperties.DISABLE_HOST_VERIFICATION, "true");
+    boolean failureCaught = false;
+    try {
+      updateClient(connectionProps);
+    } catch (Exception e) {
+      failureCaught = true;
+    }
+    assertEquals(failureCaught, true);
+  }
+
+  @Test
+  public void testSSLQuery() throws Exception {
+    final Properties connectionProps = new Properties();
+    connectionProps.setProperty(DrillProperties.ENABLE_TLS, "true");
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PATH, tsPath);
+    connectionProps.setProperty(DrillProperties.TRUSTSTORE_PASSWORD, "drill123");
+    connectionProps.setProperty(DrillProperties.DISABLE_HOST_VERIFICATION, "true");
+    try {
+      updateClient(connectionProps);
+    } catch (Exception e) {
+      TestCase.fail( new StringBuilder()
+          .append("SSL Connection failed with exception [" )
+          .append( e.getMessage() )
+          .append("]")
+          .toString());
+    }
+    test("SELECT * FROM cp.`region.json`");
+  }
+
+  @Test
+  public void testClientConfigHostnameVerification() {
+    String password = "test_password";
+    String trustStoreFileName = "drillTestTrustStore";
+    String keyStoreFileName = "drillTestKeyStore";
+    KeyStore ts, ks;
+    File tempFile1, tempFile2;
+    String trustStorePath;
+    String keyStorePath;
+
+    try {
+      String fqdn = InetAddress.getLocalHost().getHostName();
+      SelfSignedCertificate certificate = new SelfSignedCertificate(fqdn);
+
+      tempFile1 = File.createTempFile(trustStoreFileName, ".ks");
+      tempFile1.deleteOnExit();
+      trustStorePath = tempFile1.getAbsolutePath();
+      //generate a truststore.
+      ts = KeyStore.getInstance(KeyStore.getDefaultType());
+      ts.load(null, password.toCharArray());
+      ts.setCertificateEntry("drillTest", certificate.cert());
+      // Store away the truststore.
+      FileOutputStream fos1 = new FileOutputStream(tempFile1);
+      ts.store(fos1, password.toCharArray());
+      fos1.close();
```

Comment:
It's better to use try with resources thus stream will be definitely closed in case of exception.